R/utilities-qualification-gof.R

```diff
@@ -0,0 +1,409 @@
+#' @title getQualificationGOFPlotData
+#' @description Build dataframes and metadata for each GOF plot
+#' @param configurationPlan A `ConfigurationPlan` object
+#' @return plotGOFdata, a list of lists of the form list(dataframe,metadata) specific to each GOF plot
+getQualificationGOFPlotData <- function(configurationPlan) {
+  plotGOFdata <- list()
+  for (plt in seq_along(configurationPlan$plots$GOFMergedPlots)) {
+    gofPlotConfiguration <- configurationPlan$plots$GOFMergedPlots[[plt]]
+
+
+    plotGOFDataframe <- NULL
+    plotGOFMetadata <- list()
+    plotGOFMetadata$title <- gofPlotConfiguration$Title
+    plotGOFMetadata$sectionID <- gofPlotConfiguration$SectionId
+    plotGOFMetadata$plotTypes <- ospsuite::toPathArray(gofPlotConfiguration$PlotType)
+
+
+    plotGOFMetadata$axesSettings <- lapply(plotGOFMetadata$plotTypes, function(pltType) {
+      getAxesSettings(configurationPlan$plots$AxesSettings[[gofPlotAxesSettings[[pltType]]]])
+    })
+    names(plotGOFMetadata$axesSettings) <- plotGOFMetadata$plotTypes
+
+    plotGOFMetadata$groups <- list()
+
+
+    for (group in seq_along(gofPlotConfiguration$Groups)) {
+      gofPlotGroup <- gofPlotConfiguration$Groups[[group]]
+      caption <- gofPlotGroup$Caption
+      symbol <- gofPlotConfiguration$Groups[[group]]$Symbol
+      outputMappings <- gofPlotGroup$OutputMappings
+
+      plotGOFMetadata$groups[[caption]] <- list()
+      plotGOFMetadata$groups[[caption]]$outputMappings <- list()
+      plotGOFMetadata$groups[[caption]]$symbol <- symbol
+
+
+      for (omap in seq_along(outputMappings)) {
```

Comment:
Same here.
Besides, if you end up using seq_along, could you call the variable `mappingIndex` instead of `omap`, which helps identifying that the variable should be an integer (sorry if I am picky about the naming)

Reply:
Changed to `mappingIndex` since the actual index is useful for later on